### PR TITLE
Zohaib/lp 493

### PR DESCRIPTION
**Pull request by MZohaibAsaad**
### Description

[LEARNER-XXXX](https://openedx.atlassian.net/browse/LEARNER-XXXX)

Add a description of your changes here.

### Notes
- Example: Use example.sandbox.edx.org to test against
- Example: This PR will not address x,y, and z, because of a, b, and c.
- Example: Feature is behind a flag
- Example: Why I didn't add any tests

### Testing
- [ ] i18n
- [ ] RTL
- [ ] safecommit shows 0 violations
- [ ] Unit, integration, acceptance tests as appropriate
- [ ] Analytics
- [ ] Performance
- [ ] Database migrations are backwards-compatible

### CODE REVIEW

File: OpenEdXMobile/res/layout/fragment_whats_new_item.xml [hunk 1]
```diff
@@ -5,53 +5,84 @@
...
+            <ImageView
+                android:id="@+id/device"
+                android:layout_width="wrap_content"
+                android:layout_height="wrap_content"
+                android:layout_gravity="bottom|center_horizontal"
+                android:src="@drawable/device"
+                tools:ignore="ContentDescription"/>
+
 
             <ImageView
```
Comment (farhan) [inline OpenEdXMobile/res/layout/fragment_whats_new_item.xml]: This is an extra image view, using only 1 image (device frame + whats new content) is a better approach considering multi-resolution support in it.

File: OpenEdXMobile/res/layout/fragment_whats_new_item.xml [hunk 2]
```diff
@@ -5,53 +5,84 @@
...
                 android:id="@+id/image"
-                android:layout_width="match_parent"
-                android:layout_height="match_parent"
-                android:layout_gravity="bottom"
-                android:paddingLeft="@dimen/whats_new_image_horizontal_margin"
-                android:paddingRight="@dimen/whats_new_image_horizontal_margin"
+                android:layout_width="wrap_content"
+                android:layout_height="wrap_content"
+                android:layout_gravity="bottom|center_horizontal"
+                android:layout_marginTop="70dp"
```
Comment (farhan) [inline OpenEdXMobile/res/layout/fragment_whats_new_item.xml]: Avoid to hard code dimensions in layout xmls

File: OpenEdXMobile/res/values/strings.xml [hunk 3]
```diff
@@ -619,4 +619,7 @@
     <string name="rating_bar_selection">{rating} out of {num_of_stars}</string>
     <!-- Title of the Whats New screen e.g. Version 2.8.1 New Features -->
     <string name="whats_new_title">Version {version_number} New Features</string>
+
+    <string name="view_my_courses">View My Courses</string>
+    <string name="next">Next</string>
```
Comment (farhan) [inline OpenEdXMobile/res/values/strings.xml]: This should be move to labels.xml

File: OpenEdXMobile/src/main/java/org/edx/mobile/view/custom/IndicatorController.java [hunk 4]
```diff
@@ -16,14 +16,11 @@
 
 public class IndicatorController {
     private static final int FIRST_PAGE_NUM = 0;
-    private static final int DEFAULT_COLOR = 1;
```
Comment (farhan) [inline OpenEdXMobile/src/main/java/org/edx/mobile/view/custom/IndicatorController.java:19]: Good code cleaning 👍

File: OpenEdXMobile/src/main/java/org/edx/mobile/whatsnew/WhatsNewFragment.java [hunk 5]
```diff
@@ -95,10 +97,11 @@ public void onPageScrolled(int position, float positionOffset, int positionOffse
...
                 public void onPageSelected(int position) {
                     indicatorController.selectPosition(position);
-                    if (position == noOfPages - 1) {
-                        binding.doneBtn.setVisibility(View.VISIBLE);
+                    isLastPage = position == noOfPages - 1;
+                    if (isLastPage) {
+                        binding.doneBtn.setText(R.string.view_my_courses);
                     } else {
-                        binding.doneBtn.setVisibility(View.GONE);
+                        binding.doneBtn.setText(R.string.next);
```
Comment (farhan) [inline OpenEdXMobile/src/main/java/org/edx/mobile/whatsnew/WhatsNewFragment.java]: doneBtn should be renamed to nextBtn to make the code more understandable according to design.

File: OpenEdXMobile/res/raw-es/whats_new.json [hunk 6]
```diff
@@ -1,55 +1,34 @@
...
+  {
+    "version": "2.11",
+    "messages": [
+      {
+        "platforms": [
+          "android"
+        ],
+        "image": "",
+        "title": "Bienvenido",
+        "heading": "Bienvenido a la aplicación \nPhilanthropy University Content Viewer",
```
Comment (farhan) [inline OpenEdXMobile/res/raw-es/whats_new.json]: Make sure Spanish translation is correct.

File: OpenEdXMobile/src/main/java/org/edx/mobile/whatsnew/WhatsNewItemModel.java [hunk 7]
```diff
@@ -39,6 +39,8 @@ public static boolean isSupportedPlatform(@NonNull String platform) {
     @NonNull
     private String title;
     @NonNull
+    private String heading;
```
Comment (farhan) [inline OpenEdXMobile/src/main/java/org/edx/mobile/whatsnew/WhatsNewItemModel.java]: `messageTitle` seems better name.

**Comment by farhan** [comment]
App package name needs to be changed.

File: OpenEdXMobile/src/main/java/org/edx/mobile/whatsnew/WhatsNewFragment.java [hunk 8]
```diff
@@ -132,8 +135,13 @@ public void onClick(View view) {
         binding.doneBtn.setOnClickListener(new View.OnClickListener() {
             @Override
             public void onClick(View view) {
-                environment.getAnalyticsRegistry().trackWhatsNewSeen(BuildConfig.VERSION_NAME, noOfPages);
-                getActivity().finish();
+                if(isLastPage) {
+                    environment.getAnalyticsRegistry().trackWhatsNewSeen(BuildConfig.VERSION_NAME, noOfPages);
+                    getActivity().finish();
+                }
+                else{
```
Comment (farhan) [inline OpenEdXMobile/src/main/java/org/edx/mobile/whatsnew/WhatsNewFragment.java]: Code needs to be indented. 
Pro tip: Select the function and pre Command + Option key.

File: OpenEdXMobile/src/main/java/org/edx/mobile/whatsnew/WhatsNewFragment.java [hunk 9]
```diff
@@ -46,6 +46,7 @@
 
     private int noOfPages = 0;
     private int totalPagesViewed = 1;
+    private boolean isLastPage;
```
Comment (farhan) [inline OpenEdXMobile/src/main/java/org/edx/mobile/whatsnew/WhatsNewFragment.java]: Introducing a new variable is not a good approach when you can calculate it through some utility function by using `viewPager.getCurrentItem()` function.

File: OpenEdXMobile/res/layout/fragment_whats_new_item.xml [hunk 10]
```diff
@@ -5,53 +5,84 @@
...
-            android:layout_weight="0.7"
-            android:background="@drawable/whats_new_gradient">
+            android:layout_weight="0.7">
+
+            <View
+                android:layout_gravity="top"
+                android:background="@color/edx_brand_primary_base"
+                android:layout_width="match_parent"
+                android:layout_height="80dp">
+            </View>
```
Comment (farhan) [inline OpenEdXMobile/res/layout/fragment_whats_new_item.xml]: </View> Could be neglected.

File: OpenEdXMobile/res/layout/fragment_whats_new_item.xml [hunk 11]
```diff
@@ -5,53 +5,84 @@
     <LinearLayout
```
Comment (farhan) [inline OpenEdXMobile/res/layout/fragment_whats_new_item.xml:5]: avoid hardcodings in the file.

**Review by farhan — CHANGES_REQUESTED** [review]
Done with the pass.

File: OpenEdXMobile/src/main/java/org/edx/mobile/whatsnew/WhatsNewItemModel.java [hunk 12]
```diff
@@ -8,38 +8,21 @@
 import java.util.List;
 
 public class WhatsNewItemModel implements Parcelable {
-    public enum Platform {
```
Comment (farhan) [inline OpenEdXMobile/src/main/java/org/edx/mobile/whatsnew/WhatsNewItemModel.java]: enum shouldn't be declared on the start of the class ?

**Review by farhan — APPROVED** [review]
@MZohaibAsaad just one question, otherwise 
LGTM! 👍

File: OpenEdXMobile/src/main/java/org/edx/mobile/whatsnew/WhatsNewItemModel.java (hunk 12 above)
Comment (MZohaibAsaad) [reply OpenEdXMobile/src/main/java/org/edx/mobile/whatsnew/WhatsNewItemModel.java]: It was auto refactored before commit by android studio. I've reverted it back and pushed

**Pull request merged**